pkg/operators/machine-api-operator.go

```diff
@@ -32,100 +32,112 @@ var _ = Describe(
+			Expect(framework.DeleteDeployment(ctx, client, initialDeployment)).NotTo(HaveOccurred(),
+				fmt.Sprintf("Failed to delete %s Deployment", maoManagedDeployment))
 
 			By(fmt.Sprintf("checking deployment %q is available again", maoManagedDeployment))
-			Expect(framework.IsDeploymentAvailable(ctx, client, maoManagedDeployment, framework.MachineAPINamespace)).To(BeTrue())
+			Expect(framework.IsDeploymentAvailable(ctx, client, maoManagedDeployment, framework.MachineAPINamespace)).To(BeTrue(),
+				fmt.Sprintf("Failed to wait for %s Deployment to become available", maoManagedDeployment))
 
 			By(fmt.Sprintf("checking deployment %q spec matches", maoManagedDeployment))
-			Expect(framework.IsDeploymentSynced(ctx, client, initialDeployment, maoManagedDeployment, framework.MachineAPINamespace)).To(BeTrue())
+			Expect(framework.IsDeploymentSynced(ctx, client, initialDeployment, maoManagedDeployment, framework.MachineAPINamespace)).To(BeTrue(),
+				fmt.Sprintf("Failed verifying %s Deployment spec has been reconciled", maoDeployment))
 		})
 
 		It("maintains deployment spec", func() {
 			ctx := framework.GetContext()
 			client, err := framework.LoadClient()
-			Expect(err).NotTo(HaveOccurred())
+			Expect(err).NotTo(HaveOccurred(), "Failed to load client")
 
 			initialDeployment, err := framework.GetDeployment(ctx, client, maoManagedDeployment, framework.MachineAPINamespace)
-			Expect(err).NotTo(HaveOccurred())
+			Expect(err).NotTo(HaveOccurred(), fmt.Sprintf("Failed to get %s Deployment", maoManagedDeployment))
 
 			By(fmt.Sprintf("checking deployment %q is available", maoManagedDeployment))
-			Expect(framework.IsDeploymentAvailable(ctx, client, maoManagedDeployment, framework.MachineAPINamespace)).To(BeTrue())
+			Expect(framework.IsDeploymentAvailable(ctx, client, maoManagedDeployment, framework.MachineAPINamespace)).To(BeTrue(),
+				fmt.Sprintf("Failed to wait %s Deployment to become available", maoManagedDeployment))
 
 			changedDeployment := initialDeployment.DeepCopy()
 			changedDeployment.Spec.Replicas = pointer.Int32(0)
 
 			By(fmt.Sprintf("updating deployment %q", maoManagedDeployment))
-			Expect(framework.UpdateDeployment(ctx, client, maoManagedDeployment, framework.MachineAPINamespace, changedDeployment)).NotTo(HaveOccurred())
+			Expect(framework.UpdateDeployment(ctx, client, maoManagedDeployment, framework.MachineAPINamespace, changedDeployment)).NotTo(HaveOccurred(),
+				fmt.Sprintf("Failed to update %s Deployment", maoManagedDeployment))
 
 			By(fmt.Sprintf("checking deployment %q spec matches", maoManagedDeployment))
-			Expect(framework.IsDeploymentSynced(ctx, client, initialDeployment, maoManagedDeployment, framework.MachineAPINamespace)).To(BeTrue())
+			Expect(framework.IsDeploymentSynced(ctx, client, initialDeployment, maoManagedDeployment, framework.MachineAPINamespace)).To(BeTrue(),
+				fmt.Sprintf("Failed verifying %s Deployment spec has been reconciled", maoDeployment))
```

Comment:
should this be `maoManagedDeployment` ?

```suggestion
				fmt.Sprintf("Failed verifying %s Deployment spec has been reconciled", maoManagedDeployment))
```

Reply:
Thanks!